### PR TITLE
adding test for listing projects

### PR DESCRIPTION
**Pull request by Chinikins**
test added for command `spin project list`

### CODE REVIEW

File: cmd/project/list_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+func TestProjectList_malformed(t *testing.T) {
+	ts := testGateProjectList(true)
+	defer ts.Close()
+
+	rootCmd, options := cmd.NewCmdRoot(ioutil.Discard, ioutil.Discard)
+	rootCmd.AddCommand(NewProjectCmd(options))
+
+	args := []string{"project", "list", "--gate-endpoint=" + ts.URL}
+	rootCmd.SetArgs(args)
+	err := rootCmd.Execute()
```
Comment (kevinawoo) [inline cmd/project/list_test.go:53]: a malformed payload should return some kind of error message that we'd prob wanna check against

File: cmd/project/list_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+	"github.com/spinnaker/spin/cmd"
+	"github.com/spinnaker/spin/util"
+)
+
+func TestProjectList_basic(t *testing.T) {
+	ts := testGateProjectList(false)
+	defer ts.Close()
+
+	rootCmd, options := cmd.NewCmdRoot(ioutil.Discard, ioutil.Discard)
```
Comment (kevinawoo) [inline cmd/project/list_test.go:33]: pretty basic test to make sure it runs, but we could prob improve it by making sure the output of the CLI is valid json or error.

File: cmd/project/list_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	if err != nil {
+		t.Fatalf("Command failed with: %s", err)
+	}
+}
+
+func TestProjectList_malformed(t *testing.T) {
+	ts := testGateProjectList(true)
+	defer ts.Close()
+
+	rootCmd, options := cmd.NewCmdRoot(ioutil.Discard, ioutil.Discard)
```
Comment (kevinawoo) [inline cmd/project/list_test.go:48]: pretty basic test to make sure it runs, but it looks like we're not really testing the malformed payload error message

**Comment by Chinikins** [comment]
Going to close this in favor of taking on refactoring tests to follow a more robust framework.

**Pull request closed without merging**